lib/puppet/pops/validation/validator_factory_4_0.rb

```diff
@@ -7,7 +7,12 @@ class ValidatorFactory_4_0 < Factory
 
   # Produces the checker to use
   def checker diagnostic_producer
-    Checker4_0.new(diagnostic_producer)
+    if Puppet[:tasks]
+      require_relative 'tasks_checker'
+      TasksChecker.new(diagnostic_producer)
+    else
+      Checker4_0.new(diagnostic_producer)
+    end
```

Comment:
We probably want DUPLICATE_KEY to be an error - see line 35. Alternatively that we turn on strict when running with --taks... (If so, that is a separate ticket).

Reply:
I thought you had done that already in https://tickets.puppetlabs.com/browse/PUP-7921.

Reply:
yes, will add a ticket for that